bootstrap_cfn/config.py

```diff
@@ -90,6 +90,11 @@ def process(self):
         if 's3' in self.data:
             s3 = self.s3()
             map(template.add_resource, s3)
+            template.add_output(Output(
+                "StaticBucketName",
+                Description="S3 bucket name",
+                Value=Ref("StaticBucket")
+            ))
```

Comment:
Ok time to refactor here - we shouldn't be creating things outside of the `s3` function that relate to things inside it.

Options I can think of:

- lists-of-lists from `s3()`
- we could pass the Troposphere template object in to the class and let the s3 call `add_resource` or `add_output` as it needs.
- We can return a list of objects and detect where to add them. Something like:

    ```python
        objs = self.s3()
        self.add_to_template(template, objs)

    #  …

    def add_to_template(self, template, objects)
        for object in objs:
            if isinstance(obj, troposphere.Output):
                template.add_output(obj)
              elif isinstance(obj, troposphere.Parameter):
                template.add_parameter(obj)
    ```